nomad/structs/structs.go

```diff
@@ -2542,6 +2541,7 @@ type NetworkResource struct {
 	Device        string     // Name of the device
 	CIDR          string     // CIDR block of addresses
 	IP            string     // Host IP address
+	Hostname      string     `json:",omitempty"` // Hostname of the network namespace
```

Comment:
Hm, we don't often use `json` tags in here, any specific behavior you're looking for? If not I'd say let's match the common pattern of including even empty fields. It looks like we only use omitempty for old/deprecated fields at the moment.

Reply:
This was added after some discussion with @DerekStrickland that empty values should be omitted to keep the responses objects easier to understand. I feel it also makes sense in this case for the short term as the field isn't supported by all networking hooks which might add to confusion. That being said I am fine either way.

Reply:
Let's keep it for now for consistency then. While I agree the *default* should be `omitempty` I know some users appreciate the full response as a sort of template to be filled in. Sadly I'm not aware of a way to switch `omitempty` on and off at runtime with the standard Go `json` package.